pkg/oauth/apis/oauth/validation/validation_test.go

```diff
@@ -746,3 +746,38 @@ func TestValidateAuthorizeTokensUpdate(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateUserNameField(t *testing.T) {
+	fieldArg := field.NewPath("metadata", "name")
+	valid := "kube:admin"
+	errs := ValidateUserNameField(valid, fieldArg)
+	if len(errs) != 0 {
+		t.Errorf("expected success: %v", errs)
+	}
+
+	errorCases := map[string]struct {
+		name string
+		T    field.ErrorType
+		F    string
+	}{
+		"system:admin": {
+			T: field.ErrorTypeInvalid,
+			F: "metadata.name",
+		},
+	}
+	for k, v := range errorCases {
+		errs := ValidateUserNameField(k, fieldArg)
```

Comment:
please add subtests as dave explained in the aforementioned blog post. something along the lines:
```
for name, tc := range cases {
  t.Run(name, funct(t *testing.T) {
  })
}
```

`tc` is usually used as the abbreviation for test-case.

Reply:
Done